app/fakefish.py

```diff
@@ -172,21 +164,41 @@ def set_env_vars(bmc_endpoint, username, password):
     my_env["BMC_PASSWORD"] = password
     return my_env
 
-def run():
+def run(port, debug, tls_mode, cert_file, key_file):
     """
 
     """
-    app.run(host='0.0.0.0', port=port, debug=debug, ssl_context='adhoc')
+    if tls_mode == 'adhoc':
+        app.run(host='::', port=port, debug=debug, ssl_context='adhoc')
+    elif tls_mode == 'disabled':
+        app.run(host='::', port=port, debug=debug)
+    else:
+        if os.path.exists(cert_file) and os.path.exists(key_file):
+            app.run(host='::', port=port, debug=debug, ssl_context=(cert_file, key_file))
+        else:
+            app.logger.error('%s or %s not found.', cert_file, key_file)
+            exit()
 
 
 if __name__ == '__main__':
-
-    bmc_ip = os.environ.get('BMC_IP', None)
-    if bmc_ip is None:
-        app.logger.error('Configure the BMC IP using the environment variable BMC_IP')
-        exit()
+    parser = argparse.ArgumentParser(description='FakeFish, an experimental RedFish proxy that calls shell scripts for executing hardware actions.')
+    parser.add_argument('--tls-mode', type=str, choices=['adhoc', 'self-signed', 'disabled'], default='adhoc', help='Configures TLS mode. \
+                        \'self-signed\' mode expects users to configure a cert and a key files. (default: %(default)s)')
+    parser.add_argument('--cert-file', type=str, default='./cert.pem', help='Path to the certificate public key file. (default: %(default)s)')
+    parser.add_argument('--key-file', type=str, default='./cert.key', help='Path to the certificate private key file. (default: %(default)s)')
+    parser.add_argument('-r', '--remote-bmc', type=str, required=True, help='The BMC IP this FakeFish instance will connect to. e.g: 192.168.1.10')
+    parser.add_argument('-p','--listen-port', type=int, required=False, default=9000, help='The port where this FakeFish instance will listen for connections.')
+    parser.add_argument('-d', '--debug', action='store_true')
```

Comment:
nit: I tend to avoid a short variant for --debug since it shouldn't normally by used

Reply:
Changed.